src/Controller/ProxyController.php

```diff
@@ -78,6 +78,14 @@ public function legacyMetadata(string $package): Response
             ->setPublic()
             ->setLastModified((new \DateTime())->setTimestamp($metadata->timestamp()))
         ;
+
+        if ($response->isNotModified($request)) {
+            return $response;
+        }
```

Comment:
I think we can get rid of this statement since:
```php
    /**
     * Determines if the Response validators (ETag, Last-Modified) match
     * a conditional value specified in the Request.
     *
     * If the Response is not modified, it sets the status code to 304 and
     * removes the actual content by calling the setNotModified() method.
     *
     * @return bool true if the Response validators match the Request, false otherwise
     *
     * @final
     */
    public function isNotModified(Request $request): bool
```

So maybe something like this:
```php
        $response = (new StreamedResponse(ResponseCallback::fromStream($metadata->stream()), 200, [
            'Accept-Ranges' => 'bytes',
            'Content-Type' => 'application/json',
            /* @phpstan-ignore-next-line */
            'Content-Length' => fstat($metadata->stream())['size'],
        ]))
            ->setPublic()
            ->setLastModified((new \DateTime())->setTimestamp($metadata->timestamp()))
        ;
        $response->isNotModified($request);

        return $response;
```

Reply:
And please add this also to distribution method, thanks :wink:

Reply:
Added to distribution & the overall package list (the four places in the project that the cache is public)